dvc/remote/http.py

```diff
@@ -100,7 +163,13 @@ def _request(self, method, url, **kwargs):
         kwargs.setdefault("timeout", self.REQUEST_TIMEOUT)
 
         try:
-            res = self._session.request(method, url, **kwargs)
+            res = self._session.request(
```

Comment:
@pmrowla, I'm unable to make it work with Digest auth (I'm using the script that you provided on the gist).

Reply:
But, when I change this to
```suggestion
res = requests.Session().request(
```
it works. Something's wrong in our cached session perhaps?

Reply:
I think I figured it out. When I try to push, we send a `HEAD` request, after which the script sets a cookie. And, then all `POST` requests fails. So, I first tried clearing cookies with `self._session.cookies.clear()` and it worked.

Reply:
Probably, we need to set auth when on creating sessions?

Reply:
I'm wondering if this is a side effect of how my test app in the gist was configured?

By default, flask uses client-side cookies for session data, so flask-httpauth does as well. But for a web server running digest auth to be properly secured, it needs to be handled server-side: https://flask-httpauth.readthedocs.io/en/latest/#security-concerns-with-digest-authentication

If I modify the test app to actually use server-side sessions (https://gist.github.com/pmrowla/0615f162d1308cab4f429b6efafe276a) the existing remote code works without needing to clear any cached cookies

If I set `session.auth` in the http remote at the time we first create the session instead of setting it per request, I still see the same issue making requests against the original test app. So I'm not sure if it's just that `requests.auth.HTTPDigestAuth` won't work properly when talking to an improperly configured flask apps?

Reply:
> If I set session.auth in the http remote at the time we first create the session instead of setting it per request, I still see the same issue making requests against the original test app.

@pmrowla, I tried that too but, does not work. It's unclear what's the best thing to do here. 

If it's probably only `flask-httpauth`, let's ignore for now then.